src/luxonis_ml/utils/filesystem.py

```diff
@@ -110,14 +114,95 @@ def put_file(
                 else:
                     raise KeyError("No active mlflow_instance provided.")
             else:
+                import mlflow
+
                 client = mlflow.MlflowClient(tracking_uri=self.tracking_uri)
```

Comment:
Remind me again why we sometimes already have a mlflow client and sometimes not?

Reply:
CC @klemen1999. I moved the import statement within the conditionals to not have mlflow as a requirement in luxonis_ml/utils. But we can also just make it a requirement.

Reply:
The reson for this is to support live runs (like saving checkpoint at the end to current run). When there is already an mlflow instance active (in this is active in tracker) I wasn't able to connect to it by creating an MLFlowClient. But I'll look into it again now that I'm more familiar with this.